src/host_triple.rs

```diff
@@ -0,0 +1,56 @@
+use crate::emoji;
+use anyhow::Result;
+use guess_host_triple::guess_host_triple;
+use std::str::FromStr;
+use strum::Display;
+
+#[derive(Display, Debug, Clone)]
+pub enum HostTriple {
+    /// 64-bit Linux
+    #[strum(serialize = "x86_64-unknown-linux-gnu")]
+    X86_64UnknownLinuxGnu = 0,
+    /// ARM64 Linux
+    #[strum(serialize = "aarch64-unknown-linux-gnu")]
+    Aarch64UnknownLinuxGnu,
+    /// 64-bit MSVC
+    #[strum(serialize = "x86_64-pc-windows-msvc")]
+    X86_64PcWindowsMsvc,
+    /// 64-bit MinGW
+    #[strum(serialize = "x86_64-pc-windows-gnu")]
+    X86_64PcWindowsGnu,
+    /// 64-bit macOS
+    #[strum(serialize = "x86_64-apple-darwin")]
+    X86_64AppleDarwin,
+    /// ARM64 macOS
+    #[strum(serialize = "aarch64-apple-darwin")]
+    Aarch64AppleDarwin,
+}
+
+impl FromStr for HostTriple {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "x86_64-unknown-linux-gnu" => Ok(HostTriple::X86_64UnknownLinuxGnu),
+            "aarch64-unknown-linux-gnu" => Ok(HostTriple::Aarch64UnknownLinuxGnu),
+            "x86_64-pc-windows-msvc" => Ok(HostTriple::X86_64PcWindowsMsvc),
+            "x86_64-pc-windows-gnu" => Ok(HostTriple::X86_64PcWindowsGnu),
+            "x86_64-apple-darwin" => Ok(HostTriple::X86_64AppleDarwin),
+            "aarch64-apple-darwin" => Ok(HostTriple::Aarch64AppleDarwin),
+            _ => Err(format!(
+                "{} Host triple '{}' is not supported.",
+                emoji::ERROR,
+                s
+            )),
+        }
+    }
+}
```

Comment:
This is fine of course, but you can also derive the `strum::EnumString` attribute to get this automatically:  
https://docs.rs/strum_macros/latest/strum_macros/derive.EnumString.html

Reply:
Ahh okay, that's fair. You can probably call `.map_err()` on the result but if that becomes too tedious it probably makes sense to just leave this version.

Reply:
I've tried using `.map_err()` but I was having a lot of type errors as strum uses [`ParseError`](https://docs.rs/strum/0.11.0/strum/enum.ParseError.html). I've found a way to leverage `EnumString` implementations while having a verbose error. Is not perfect, but I think it's an improvement. 

Let me know what you think, and once it's polished, I'll do the same for [tagets.rs](https://github.com/esp-rs/espup/blob/feature/default-host-arg/src/targets.rs#L24-L36)

Reply:
TIL/Today I Remembered `anyhow::context`. I think `context` improves the previous solution as you don't need two nested `match` arguments.